lib/Echidna/Campaign.hs

```diff
@@ -395,20 +396,32 @@ callseq vm txSeq = do
     -> (FunctionName -> Maybe AbiType)
     -> Map AbiType (Set AbiValue)
   returnValues txResults returnTypeOf =
-    Map.fromList . flip mapMaybe txResults $ \(tx, result) -> do
-      case result of
+    Map.unionsWith Set.union . mapMaybe extractValues $ txResults
+    where
+      extractValues (tx, result) = case result of
         VMSuccess (ConcreteBuf buf) -> do
           fname <- case tx.call of
             SolCall (fname, _) -> Just fname
             _ -> Nothing
           type' <- returnTypeOf fname
           case runGetOrFail (getAbi type') (LBS.fromStrict buf) of
-            -- make sure we don't use cheat codes to form fuzzing call sequences
-            Right (_, _, abiValue) | abiValue /= AbiAddress (forceAddr cheatCode) ->
-              Just (type', Set.singleton abiValue)
+            Right (_, _, abiValue) ->
+              if isTuple type'
+                then Just $ Map.fromListWith Set.union
+                      [ (abiValueType val, Set.singleton val)
+                      | val <- V.toList $ getTupleVector abiValue
```

Comment:
should these be filtered so no `cheatCode` makes its way into the dictionary?

Reply:
Should we move the cheat code detection deeper into the dictionary handling to avoid duplication?